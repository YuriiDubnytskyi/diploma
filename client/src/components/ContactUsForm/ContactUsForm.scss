.contact__conatiner {
    width: 80%;
    margin: 30px auto;
}
.contact-user {
    display: flex;
}
.contact-user-name,
.contact-user-email {
    width: 50%;
    display: flex;
    flex-direction: column;
}
.input--contact-name {
    width: 90%;
}
.contact-subject,
.contact-text {
    width: 100%;
    display: flex;
    flex-direction: column;
}
.input--contact-text {
    height: 150px;
}
.input {
    background-color: rgb(153, 153, 153);
    color: white;
    padding: 10px;
    border: none;
}
.contact-user,
.contact-subject,
.contact-text {
    margin: 20px 0;
}
.contact-btn {
    width: 99px;
    border: 1px solid #00c48c;
    text-align: center;
    padding: 10px 20px;
    color: #ffffff;
    background-color: #00c48c;
    transition: ease-in 0.3s;
}
.contact-btn:hover {
    color: #00c48c;
    border: 1px solid #00c48c;
    border-radius: 25px;
    font-weight: 700;
    cursor: pointer;
    background-color: #ffffff;
}
.contact-info {
    display: flex;
}
.contact-info-success,
.contact-info-error {
    text-align: center;
    padding: 10px 20px;
    color: #ffffff;
    margin: 0 20px;
}
.contact-info-success {
    background-color: #00c48c;
    width: 100px;
}
.contact-info-error {
    background-color: #881111;
}
@media screen and (max-width: 600px) {
    .contact-info {
        flex-direction: column-reverse;
        align-items: center;

        .contact-btn {
            margin: 10px;
        }
    }
    .contact-user {
        flex-direction: column;
    }
    .contact-user-name,
    .contact-user-email {
        width: 100%;
    }
    .input--contact-name {
        width: 95%;
    }
}
