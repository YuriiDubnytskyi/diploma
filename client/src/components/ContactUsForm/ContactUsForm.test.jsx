import React from "react";
import ContactUsForm from "./ContactUsForm";

const setUp = (props) => shallow(<ContactUsForm {...props} />);

describe("Test HTML in Conact Form", () => {
    it("Should contain four input fields", () => {
        const component = setUp();
        const wrapper = component.find(".input");
        expect(wrapper.length).toBe(4);
    });

    it("Contact Form without error or success", () => {
        const component = setUp({ success: "", err: "" });
        expect(component).toMatchSnapshot();
    });

    it("Should contain .contact-info-success", () => {
        const component = setUp({ success: "Success" });
        const wrapper = component.find(".contact-info-success");
        expect(wrapper.length).toBe(1);
    });

    it("Should contain .contact-info-error", () => {
        const component = setUp({ err: "Error" });
        const wrapper = component.find(".contact-info-error");
        expect(wrapper.length).toBe(1);
    });
});
