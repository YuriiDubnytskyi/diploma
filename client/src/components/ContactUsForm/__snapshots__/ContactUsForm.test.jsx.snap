// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HTML in Conact Form Contact Form without error or success 1`] = `
<div
  className="contact__conatiner contact"
>
  <div
    className="contact__user contact-user"
  >
    <div
      className="contact-user-name"
    >
      <label>
        Ваше Імя
      </label>
      <input
        className="input input--contact-name"
      />
    </div>
    <div
      className="contact-user-email"
    >
      <label>
        Ваша пошта
      </label>
      <input
        className="input input--contact-email"
      />
    </div>
  </div>
  <div
    className="contact-subject"
  >
    <label>
      Тема
    </label>
    <input
      className="input input--contact-subject"
    />
  </div>
  <div
    className="contact-text"
  >
    <label>
      Повідомлення
    </label>
    <textarea
      className="input input--contact-text"
    />
  </div>
  <div
    className="contact-info"
  >
    <p
      className="contact-btn"
    >
      Надіслати
    </p>
  </div>
</div>
`;
